### DISCUSSION TITLE
How to use application specific password when member of multiple teams

### DISCUSSION THREAD

**Discussion by flatpackcode2** [fastlane/fastlane#17478] [Q&A]
Hi there,

If we are a developer on multiple Apple Teams and would like to use app-specific-password as part of the upload to the appstore/testflight, do we need to get the acct holder of each team to create an app-specific password?

My issue is this: From the docs, I understand that we set the app-specific password in the env var in Circle Ci (for example). However, if I'm pushing builds to multiple teams in parallel this would mean that at any given moment, only one build can go through because I can only set one team's FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD at a given moment.

Did I understand this correctly? And if so, does the AppStoreConnet API overcome this?

**Comment by rogerluan** [comment]
Hey @flatpackcode2 ! 👋 

>do we need to get the acct holder of each team to create an app-specific password?

Yes!

However, I would personally discourage harassing them to do so at this point, because app-specific passwords don't work with everything (only a certain subset of tasks - for the rest of them, you'll need a full authenticated session). So I'd harass them to provide you an API auth token 😁 You can read more about it here https://docs.fastlane.tools/app-store-connect-api/

You can then pass arguments on a per-command-basis for each of your builds, to make sure your environment variables don't clash.
